Gems/AtomTressFX/Assets/Shaders/HairLightTypes.azsli

```diff
@@ -0,0 +1,633 @@
+void ApplyPointLight(ViewSrg::PointLight light, Surface surface, inout LightingData lightingData)
+{
+    float3 posToLight = light.m_position - surface.position;
+    float d2 = dot(posToLight, posToLight); // light distance squared
+    float falloff = d2 * light.m_invAttenuationRadiusSquared;
+
+    // Only calculate shading if light is in range
+    if (falloff < 1.0f)
+    {
+        // Smoothly adjusts the light intensity so it reaches 0 at light.m_attenuationRadius distance
+        float radiusAttenuation = 1.0 - (falloff * falloff);
+        radiusAttenuation = radiusAttenuation * radiusAttenuation;
+
+        // Standard quadratic falloff
+        d2 = max(0.001 * 0.001, d2); // clamp the light to at least 1mm away to avoid extreme values.
+        float3 lightIntensity = (light.m_rgbIntensityCandelas / d2) * radiusAttenuation;
+
+        float3 dirToLight = normalize(posToLight);
+        SetNormal(surface.tangent, dirToLight, surface, lightingData);
+
+        // Diffuse contribution
+        lightingData.diffuseLighting += GetDiffuseLighting(surface, lightingData, lightIntensity, dirToLight);
+
+        // Tranmission contribution
+        lightingData.translucentBackLighting += GetHairBackLighting(surface, lightingData, lightIntensity, dirToLight, 1.0);
+
+        // Adjust the light direcion for specular based on bulb size
+
+        // Calculate the reflection off the normal from the view direction
+        float3 reflectionDir = reflect(-lightingData.dirToCamera, surface.normal);
+
+        // Calculate a vector from the reflection vector to the light
+        float3 reflectionPosToLight = posToLight - dot(posToLight, reflectionDir) * reflectionDir;
+
+        // Adjust the direction to light based on the bulb size
+        posToLight -= reflectionPosToLight * saturate(light.m_bulbRadius / length(reflectionPosToLight));
+
+        // Adjust the intensity of the light based on the bulb size to conserve energy
+        float sphereIntensityNormalization = GetIntensityAdjustedByRadiusAndRoughness(surface.roughnessA, light.m_bulbRadius, d2);
+
+        // Specular contribution
+        lightingData.specularLighting += sphereIntensityNormalization * GetSpecularLighting(surface, lightingData, lightIntensity, normalize(posToLight));
+    }
+}
+
+void ApplyAllPointLights(Surface surface, inout LightingData lightingData)
+{
+    for (int l = 0; l < ViewSrg::m_pointLightCount; ++l)
+    {
+        ViewSrg::PointLight light = ViewSrg::m_pointLights[l];
+        ApplyPointLight(light, surface, lightingData);
+    }
+}
+
+//==============================================================================
+//                             Disk Lights
+//==============================================================================
+#include <Atom/Features/PBR/Lights/DiskLight.azsli>
+
+void TestApplyDiskLight(ViewSrg::DiskLight light, Surface surface, inout LightingData lightingData)
```

Comment:
Why "Test" in the name here? If it's intentional can you add a comment explaining why this one is test and the others are not?

Reply:
great catch.  It was due to the inclusion of the azsli, however, since we now implemented all the lighting shader options, I tested and the original method is working including shadowing and so I'm removing this piece of code.